tests/user_input_checks_test.py

```diff
@@ -490,3 +490,55 @@ def test_validate_theta_and_x_gpu():
 
     with pytest.raises(AssertionError) as exc:
         validate_theta_and_x(theta, x.to(torch.float64))
+
+
+@pytest.mark.gpu
+@pytest.mark.parametrize(
+    "user_simulator, user_prior",
+    (
+        (
+            diagonal_linear_gaussian,
+            BoxUniform(zeros(3, dtype=torch.float64), ones(3, dtype=torch.float64)),
+        ),
+        (linear_gaussian_no_batch, BoxUniform(zeros(3), ones(3))),
+        (
+            numpy_linear_gaussian,
+            UserNumpyUniform(zeros(3), ones(3), return_numpy=True),
+        ),
+        (diagonal_linear_gaussian, BoxUniform(zeros(3), ones(3))),
+        (linear_gaussian_no_batch, BoxUniform(zeros(3), ones(3))),
+        (list_simulator, BoxUniform(-ones(3), ones(3))),
+        (
+            numpy_linear_gaussian,
+            UserNumpyUniform(zeros(3), ones(3), return_numpy=True),
+        ),
+        (
+            diagonal_linear_gaussian,
+            (
+                Gamma(ones(1), ones(1)),
+                Beta(ones(1), ones(1)),
+                MultivariateNormal(zeros(2), eye(2)),
+            ),
+        ),
+    ),
+)
```

Comment:
we would need to test this just for a single (arbitrary) instance of a simple prior and simulator, no?

Reply:
right, to be frank, I copy & pasted this from another test in order to get started. But you are absolutely right, this is overkill.

Reply:
I adressed this in 2d4fe80